docs/en/Community-Articles/2023-03-20- IdentityServer-Relationships/Post.md

```diff
@@ -0,0 +1,239 @@
+## Introduction
+In this article, I will talk about the relationships of IdentityServer in every web application you create using ABP framework. When you read this article, you will learn how to extend the user entity in applications you develop using the ABP framework with a primitive type, extending the user by associating the user with another entity (User-many-to-one-X).
+
+
+## Creating the Solution 
+>For the source code of the application: https://github.com/onurpicakci/Abp-Identity-Relationship
+
+In this article we will use EF Core as the database and MVC as the user interface. But Angular will also work on Blazor Server and Blazor WebAssembly. Abp Framework offers starter templates to get started faster. We can create a new starter template using the Abp CLI:
+
+```
+abp new IdentityRelationship
+```
+
+After the download is complete, you can run the `IdentityRelationship.DbMigrator` project to create database migrations. You can then run the `IdentityRelationship.Web` project to see our application running.
+
+> Default admin username is **admin** and password is **1q2w3E\***
+
+![solution-image](images/solution-image.png)
+
+## Module Entity Extensions
+
+Module entity extension system is a high level extension system that allows you to define new properties for existing entities of the depended modules. It automatically adds properties to the entity, database, HTTP API and the user interface in a single point.
+
+### Extending the User Entity With a Primitive Type
+
+Open the `IdentityRelationshipModuleExtensionConfigurator` class inside the `Domain.Shared` project of your solution and change the `ConfigureExtraPropertiesmethod` as shown below to add a `IdentificationNumber property` to the `IdentityUser` entity of the [Identity Module](https://docs.abp.io/en/abp/latest/Modules/Identity).
+
+```csharp
+public static void ConfigureExtraProperties()
+{
+    OneTimeRunner.Run(() =>
+    {
+        ObjectExtensionManager.Instance.Modules()
+            .ConfigureIdentity(identity =>
+            {
+                identity.ConfigureUser(user =>
+                {
+                    user.AddOrUpdateProperty<string>( //property type: string
+                        "IdentificationNumber", //property name
+                        property =>
+                        {
+                            //validation rules
+                            property.Attributes.Add(new RequiredAttribute());
```

Comment:
I recommend not using `RequiredAttribute` here. I know this is an article but some developers will copy this code directly, rename it and use it in their own projects. However, since this item is required in the user table, a new user will encounter a null exception while registering.

See: 

> when i flow you to change the AppUser info, the UI is running ok,but i cann't Register a new user ,it's still to show error for Title Null Exception.
_ref: https://community.abp.io/posts/how-to-add-custom-property-to-the-user-entity-6ggxiddr (comments)_